resin/logs.py

```diff
@@ -105,4 +106,12 @@ def get_channel(uuid):
 
         """
 
-        return 'device-{uuid}-logs'.format(uuid=uuid)
+        if not hasattr(self, 'logs_channel'):
+            device_info = self.device.get(uuid)
+            print device_info
+            if 'logs_channel' in device_info:
+                self.logs_channel = device_info['logs_channel']
+            else:
+                self.logs_channel = uuid
+
+        return 'device-{logs_channel}-logs'.format(logs_channel=self.logs_channel)
```

Comment:
This need not be so verbose--it's a single variable. Maybe use the {0} shorthand?